android/engine/src/main/java/org/smartregister/fhircore/engine/util/SecureSharedPreference.kt

```diff
@@ -72,6 +71,19 @@ class SecureSharedPreference @Inject constructor(@ApplicationContext val context
     secureSharedPreferences.edit { putString(KEY_SESSION_PIN, pin) }
   }
 
+  fun savePinCredentials() {
+    secureSharedPreferences.edit {
+      putString(KEY_PIN_CREDENTIALS, retrieveCredentials().encodeJson())
+      putString(KEY_LATEST_SESSION_TOKEN_PREFERENCE, retrieveCredentials()!!.sessionToken)
+    }
+  }
+
+  fun retrievePinCredentials(): AuthCredentials? {
+    return secureSharedPreferences
+      .getString(KEY_PIN_CREDENTIALS, null)
+      ?.decodeJson<AuthCredentials>()
+  }
+
```

Comment:
Does this code the latest change requests where we do not clear the shared preferences even after login to facilitate offline login?

Reply:
these changes are required for pin based login

Reply:
https://github.com/opensrp/fhircore/issues/1098